tiledb/sm/query/query.cc

```diff
@@ -2516,77 +2516,12 @@ void Query::set_status(QueryStatus status) {
 }
 
 Status Query::set_subarray(const void* subarray) {
-  if (type_ != QueryType::READ && type_ != QueryType::WRITE &&
-      type_ != QueryType::MODIFY_EXCLUSIVE) {
-    return LOG_STATUS(Status_SerializationError(
-        "Cannot set subarray; Unsupported query type."));
-  }
-
```

Comment:
Don't see the above tests still being covered in your changes, seems these particular tests were -not- RepeatingYourself, are they known to no longer be necessary?